eureka2-server/src/main/java/com/netflix/eureka2/server/service/SelfRegistrationService.java

```diff
@@ -77,8 +77,10 @@ public void call() {
                         public void call(Throwable throwable) {
                             logger.warn("Error during self unregistration process: {}", latestSelfInfo);
                         }
-                    }).materialize().toBlocking().firstOrDefault(null);
+                    }).materialize().toBlocking().firstOrDefault(null);  // FIXME why materialize and block here?
```

Comment:
We cannot/shouldn't proceed with shutdown until unregister is completed. We cannot change shutdown() to return observable, as it is a pre-destroy hook.